This sequence-to-sequence experiment aims to learn joint representation of document in one particular language (e.g. English) to document in another language (e.g. Dutch). For this first stage of experiment, we use a parallel corpus from europarl (the one available in data/europarl_*_test is a splitted sample of first paragraph from the original europarl text document - for a purpose of tutorial).

In [1]:
from text_preprocessing import *

In [3]:
from language_models import *

The data available here have been preprocessed and the pre-processing code is also available in text_preprocessing.py (the detail of pre-processing stage and tutorial is available in 'preprocessing.ipynb'). Example of pre-processed europarl data, as follows:

In [5]:
vocab = readPickle('data/europarl_en_nl_test/vocabulary')

vocabulary consists of 2 keys (python dictionary format): 
0 for English or 'en' vocabulary, which will be look-up vocabulary/dictionary for the input. 1 for Dutch or 'nl' vocabulary, which is the dictionary for text output in sequence-to-sequence model.

In [6]:
vocab.keys()

[0, 1]

In [11]:
vocab[0].items()[:5]

[(0, 'zero'), (1, 'represent'), (2, 'all'), (3, 'plenary'), (4, 'month')]

To access the word based on index:

In [12]:
vocab[0][0]

'zero'

and the following is vocabulary in Dutch (nl) language:

In [13]:
vocab[1].items()[:5]

[(0, 'zero'), (1, 'mensen'), (2, 'doel'), (3, 'Hervatting'), (4, 'inderdaad')]

The dictionary is not sorted in parallel. So, it is not a translation dictionary, but a look-up indexing words for each language.

In [14]:
len(vocab[0])

461

Since, the data is a sample from the original document, the number of word English vocabulary here is small = 461. Meanwhile, the vocabulary of dutch document = 483

In [15]:
len(vocab[1])

483

In [16]:
documents = readPickle('data/europarl_en_nl_test/documents')

In [17]:
documents.keys()

['en', 'nl']

Each language only consists of one document. For instance, in english document corpus, the sample document consists of 1197 sentences. 

In [20]:
len(documents['en'][0])

1197

While the paralel version - Dutch document consists of 1189 sentences

In [21]:
len(documents['nl'][0])

1189